worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -127,34 +127,91 @@ void processContract(Job job, Path outputDirPath)
         Contract contract = job.getContract();
         assert contract != null;
         log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
-        // Retrieve the contract beneficiaries
 
-        Map<Long, CoverageSummary> patients = processContractBenes(job);
+        try {
+            // Retrieve the contract beneficiaries
+            Map<Long, CoverageSummary> patients = processContractBenes(job);
+
+            // Create a holder for the contract, writer, progress tracker and attested date
+            JobData jobData = new JobData(job.getJobUuid(), job.getSince(), getOrganization(job), patients);
+
+            var jobOutputs = contractProcessor.process(outputDirPath, job, jobData);
+
+            // For each job output, add to the job and save the result
+            jobOutputs.forEach(job::addJobOutput);
+            jobOutputRepository.saveAll(jobOutputs);
+
+            // If the job is done searching
+            verifyTrackedJobProgress(job, contract);
+        } finally {
+            // Guarantee that we write out statistics on the job if possible
+            persistTrackedJobProgress(job, contract);
+        }
+    }
+
+    void verifyTrackedJobProgress(Job job, Contract contract) {
+        ProgressTracker progressTracker = jobProgressService.getStatus(job.getJobUuid());
+
+        if (progressTracker == null) {
+            log.info("Job [{}] - contract [{}] does not have any progress information, skipping verifying tracker",
+                    job.getJobUuid(), contract.getContractNumber());
+            return;
+        }
 
-        // Create a holder for the contract, writer, progress tracker and attested date
-        JobData jobData = new JobData(job.getJobUuid(), job.getSince(), getOrganization(job), patients);
+        // Number in database
+        int expectedPatients = progressTracker.getPatientsExpected();
 
-        var jobOutputs = contractProcessor.process(outputDirPath, job, jobData);
+        // Number queued to retrieve
+        int queuedPatients = progressTracker.getPatientRequestQueuedCount();
 
-        // For each job output, add to the job and save the result
-        jobOutputs.forEach(job::addJobOutput);
-        jobOutputRepository.saveAll(jobOutputs);
+        // Number of retrievals processed
+        int processedPatients = progressTracker.getPatientRequestProcessedCount();
 
+        if (expectedPatients != queuedPatients) {
+            String alertMessage = String.format("[%s] expected beneficiaries (%d) does not match queued beneficiaries (%d)",
+                    job.getJobUuid(), expectedPatients, queuedPatients);
+            eventLogger.alert(alertMessage, Ab2dEnvironment.PROD_LIST);
+        }
+
+        if (expectedPatients != processedPatients) {
```

Comment:
Since some errors are allowed, shouldn't this be:
if (expectedPatients != processedPatients + patientFailureCount)

Reply:
It depends on how you count processedPatients. Right now we're counting all patients whether the eob request is processed successfully or not.